internal/queryexecutor/range_query_planner.go

```diff
@@ -0,0 +1,117 @@
+package queryexecutor
+
+import (
+	"errors"
+
+	"github.com/IBM-Blockchain/bcdb-server/internal/stateindex"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/constants"
+	"github.com/IBM-Blockchain/bcdb-server/pkg/types"
+)
+
+type rangeQueryPlan struct {
+	startKey *stateindex.IndexEntry
+	endKey   *stateindex.IndexEntry
+}
+
+func createQueryPlan(attribute string, conds *attrConditions) (*rangeQueryPlan, error) {
+	// we assume this function to get only valid conditions
+	//   - eq and no other conditions
+	//   - lt and lte do not appear together
+	//   - gt and gte do not appear together
```

Comment:
It also makes sense to put that comment on the commit before this one, that validates the conditions...

Reply:
Done, already.